lib/inertia_rails/inertia_rails.rb

```diff
@@ -47,4 +47,17 @@ def self.evaluated_version
   def self.evaluated_blocks(controller,  blocks)
     blocks.flat_map { |block| controller.instance_exec(&block) }.reduce(&:merge) || {}
   end
+
+  # thread_mattr_accessor doesn't accept :default as an option, and since the method
+  # is actually defined in the context of Thread.current, we can't do something like:
+  # def self.shared_plain_data
+  #   @@shared_plain_data || {}
+  # end
+  def self.threadsafe_shared_plain_data
+    self.shared_plain_data || {}
+  end
+
+  def self.threadsafe_shared_blocks
+    self.shared_blocks || []
+  end
```

Comment:
this syntax is a bit awkward... it's not as easy to define a default value for the the thread safe `mattr_accessor`

Reply:
I think this design makes sense overall, but would it be clearer of an external api to add getters and setters and treat the internal value as private? something like

```ruby
thread_mattr_accessor :threadsafe_shared_plain_data

def self.shared_plain_data
   self.threadsafe_shared_plain_data || {}
end

def self.shared_plain_data=(val)
   self.threadsafe_shared_plain_data = val
end
```